src/control/ToolHandler.cpp

```diff
@@ -242,8 +259,18 @@ auto ToolHandler::getThickness(ToolPointer toolPointer) -> double {
     return 0;
 }
 
-void ToolHandler::setSize(ToolSize size, ToolPointer toolPointer) {
-    Tool* tool = getToolPointer(toolPointer);
+void ToolHandler::setSize(ToolSize size) {
+    Tool* tool = this->activeTool;
+    if (size < TOOL_SIZE_VERY_FINE || size > TOOL_SIZE_VERY_THICK) {
+        g_warning("ToolHandler::setSize: Invalid size! %i", size);
```

Comment:
Please clamp the value only. Somebody might switch from very low to (too) high. Best would be to still end up as high as possible in this scenario. If we get a bug here then it would be low priority instead of the high priority for not changing tool size at all, as would occur with your current code.

Reply:
Sounds good. I'll use `std::clamp` from `<algorithm>` and still raise the warning to make it not fully silent.

Reply:
Yeah exactly! It should not be fully silent but as it is only a warning it should still do mostly what the user would expect.